packages/common/src/tools/LogThrottler.ts

```diff
@@ -0,0 +1,71 @@
+interface LogInfo {
+  count: number
+  throttleCount: number
+  isThrottling: boolean
+}
+
+export class LogThrottler {
```

Comment:
I think we need tests for this. However, more importantly, I am worried that this implementation might be too resource heavy -- it creates a timer for every single thing logged. 

Maybe we can create something with a slightly different behavior but much more performant. What if we count logs only that happened last minute ie. clear `recentLogs` every minute. (This allows you to get rid of `setTimeout` for every log message). From practical PoV, I think it should work the same way.

Reply:
Yeah, I was thinking about something similar to the idea you proposed. The only drawback of this approach is that instead of checking if x logs have been called during y time, we would check if x logs have been called within y time span. But regarding the performance, this would be better and should not cause any issues.